index.html

```diff
@@ -157,114 +285,17 @@
-                        steps.push([toSwap, 'left', toFind, 'right']);
-                    } else if (neededSwaps.middle.length == 1) {
-                        // it's possible that we're handling 'all outside aisles must swap one' e.g. SCT/Cylinder+Cone+Prism
-                        // we've popped left, pop middle and see if right needs one swap. if so, we'll actually need to swap the retrieved symbol again.
-                        toFind = neededSwaps.middle.pop();
-                        steps.push([toSwap, 'left', toFind, 'middle']);
-                        if (neededSwaps.left.length === 0 && neededSwaps.middle.length === 0 && neededSwaps.right.length == 1) {
-                            neededSwaps.left.push(toFind);
-                        }
-                    } else if (neededSwaps.right.length == 1) {
-                        toFind = neededSwaps.right.pop();
-                        steps.push([toSwap, 'left', toFind, 'right']);
-                    } else {
-                        console.log('wanted to swap ' + toSwap + ' from left, but couldnt ??');
-                        steps.push([undefined, undefined, undefined, undefined]);
-                        break;
-                    }
-                } else if (neededSwaps.middle.length) {
-                    toSwap = neededSwaps.middle.pop();
-                    if (neededSwaps.right.length) {
-                        toFind = neededSwaps.right.pop();
-                        steps.push([toSwap, 'middle', toFind, 'right']);
-                    } else {
-                        console.log('wanted to swap ' + toSwap + ' from middle, couldnt put it at right?? left should have taken this otherwise??');
-                        steps.push([undefined, undefined, undefined, undefined]);
-                        break;
-                    }
-                } else {
-                    // left needs zero or two swaps,
-                    // middle needs zero or two swaps,
-                    // but if any aisle needs two swaps both others need at least one to match
-                    // so if we are here then either we are done (should have exited the loop) or all aisles need two swaps.
-                    // break the tie, at which point the loop can converge.
-                    toSwap = neededSwaps.middle.pop();
-                    toFind = neededSwaps.right.pop();
-                    steps.push([toSwap, 'middle', toFind, 'right']);
-                }
+	  getSolution: function(){
+        if(this.insideGuardianLeft && this.insideGuardianMiddle && this.insideGuardianRight && this.outsideGuardianLeft && this.outsideGuardianMiddle && this.outsideGuardianRight){
+            var scenario = this.insideGuardianLeft + this.insideGuardianMiddle + this.insideGuardianRight + this.outsideGuardianLeft + this.outsideGuardianMiddle + this.outsideGuardianRight;
+            if (!this.solutionsMatrix.hasOwnProperty(scenario)){
```

Comment:
free thought: from a solution point of view, many of the solutions here are kinda redundant - as one example, `TriangleSquareCirclePrismPrismSphere` is the same as `SquareTriangleCirclePrismPrismSphere`. if you key the solutions as `insideL + outsideL + insideM + outsideM + insideR + outsideR` you could cut down the number of solutions to look at by sorting left/mid/right before concatenating each aisle and looking up. then you'd need to remap the solution to the correct real lanes. as an example, the earlier `(S+TC, T+TC, C+SS)` example would be looked up as `(C+SS, S+TC, T+TC)`, then once you have the solution remapping that to the real S/T/C calls to fix up left/middle/right. that would drop the number of states down to 21!

it's also somewhat more complicated, so i totally understand if this just sounds unhinged. what's here seems totally reasonable :)